CenterForOpenScience/waterbutler
# [SVCS-590] Improve WB Move and Copy

## Ticket

https://openscience.atlassian.net/browse/SVCS-590

## Purpose

"Actions"  are used in different places having different means. As requested by @icereval , this PR clean up the logic by using `provider_action` and `auth_action`. There is also an `osf_action` already introduced in https://github.com/CenterForOpenScience/waterbutler/pull/288.

## Changes

### Primary

Replace `action` with `provider_action` and `auth_action`. The former are native actions supported by providers: `move` and `copy`. The latter are actions requested in the API request and used by the auth handler: `move`, `rename` and `copy`. Both `move` and `copy` in `auth_action` are mapped to `move` in `provider_action`.

### Minor

* Use `.get('key', 'default_value') instead of `['key']` for dictionary access.
* Use `http.HTTPStatus` instead of number literals

## Side effects

No

## QA Notes

Backend follow-up update after https://github.com/CenterForOpenScience/waterbutler/pull/288. Everything should word as expected. Testing a few different providers is recommended.

* move/copy/rename within a project
* move/copy/rename between two projects you own
* move/copy/rename between a project you own and a linked public project

## Deployment Notes

No
